controlplane/kubeadm/internal/controllers/controller.go

```diff
@@ -802,45 +813,169 @@ func (r *KubeadmControlPlaneReconciler) syncMachines(ctx context.Context, contro
 
 // reconcileControlPlaneConditions is responsible of reconciling conditions reporting the status of static pods and
 // the status of the etcd cluster.
-func (r *KubeadmControlPlaneReconciler) reconcileControlPlaneConditions(ctx context.Context, controlPlane *internal.ControlPlane) error {
+func (r *KubeadmControlPlaneReconciler) reconcileControlPlaneConditions(ctx context.Context, controlPlane *internal.ControlPlane) (reterr error) {
 	// If the cluster is not yet initialized, there is no way to connect to the workload cluster and fetch information
 	// for updating conditions. Return early.
-	if !controlPlane.KCP.Status.Initialized {
+	// We additionally check for the Available condition. The Available condition is set at the same time
+	// as .status.initialized and is never changed to false again. Below we'll need the transition time of the
+	// Available condition to check if the remote conditions grace period is already reached.
+	// Note: The Machine controller uses the ControlPlaneInitialized condition on the Cluster instead for
+	// the same check. We don't use the ControlPlaneInitialized condition from the Cluster here because KCP
+	// Reconcile does (currently) not get triggered from condition changes to the Cluster object.
+	controlPlaneInitialized := conditions.Get(controlPlane.KCP, controlplanev1.AvailableCondition)
```

Comment:
The ControlPlaneInitialized condition on the cluster will be also there with v1beta2 and it will also only ever go from false to true (and never back to false), correct? 

If I'm correct then:
a) the implementation in the Machine controller is future proof
b) we should adjust the KCP implementation to behave the same way (I only have to adjust the predicates we use in KCP)